src/Features/VisualBasic/Portable/CodeLens/VisualBasicDisplayInfoService.vb

```diff
@@ -0,0 +1,161 @@
+                SymbolDisplayMemberOptions.IncludeContainingType Or SymbolDisplayMemberOptions.IncludeParameters,
+                SymbolDisplayDelegateStyle.NameOnly,
+                SymbolDisplayExtensionMethodStyle.StaticMethod,
+                SymbolDisplayParameterOptions.IncludeType,
+                SymbolDisplayPropertyStyle.ShowReadWriteDescriptor,
+                SymbolDisplayLocalOptions.IncludeType,
+                SymbolDisplayKindOptions.None,
+                SymbolDisplayMiscellaneousOptions.EscapeKeywordIdentifiers Or SymbolDisplayMiscellaneousOptions.UseSpecialTypes)
+
+        Private Shared ReadOnly ShortFormat As SymbolDisplayFormat =
+            DefaultFormat.WithTypeQualificationStyle(SymbolDisplayTypeQualificationStyle.NameAndContainingTypes)
+
+        ''' <summary>
+        ''' Returns the node that should be displayed
+        ''' </summary>
+        Public Function GetDisplayNode(node As SyntaxNode) As SyntaxNode Implements ICodeLensDisplayInfoService.GetDisplayNode
+            Select Case node.Kind()
+                ' A variable declarator can contain multiple symbols, for example "Private field2, field3 As Integer"
+                ' In that case default to the first field name.
+                Case SyntaxKind.VariableDeclarator
+                    Dim variableNode = CType(node, VariableDeclaratorSyntax)
+                    Return GetDisplayNode(variableNode.Names.First())
+
+                ' A field declaration (global variable) can contain multiple symbols, for example "Private field2, field3 As Integer"
+                ' In that case default to the first field name.
+                Case SyntaxKind.FieldDeclaration
+                    Dim fieldNode = CType(node, FieldDeclarationSyntax)
+                    Return GetDisplayNode(fieldNode.Declarators.First())
+
+                Case SyntaxKind.PredefinedType
+                    Return GetDisplayNode(node.Parent)
+
+                Case SyntaxKind.DocumentationCommentTrivia
+                    If node.IsStructuredTrivia Then
+                        Dim structuredTriviaSyntax = CType(node, StructuredTriviaSyntax)
+                        Return GetDisplayNode(structuredTriviaSyntax.ParentTrivia.Token.Parent)
+                    Else
+                        Return node
+                    End If
+            End Select
+
+            Return node
+        End Function
+
+        Private Shared Function SymbolToDisplayString(symbolDisplayFormat As SymbolDisplayFormat, symbol As ISymbol) As String
+            If symbol Is Nothing Then
+                Return VBFeaturesResources.paren_unknown_paren
+            End If
+
+            Dim symbolName As String = symbol.ToDisplayString(symbolDisplayFormat)
+
+            ' surounding a idenitfier in square brackets allows you to use a keyword as an identifer
+            symbolName = symbolName.Replace("[", String.Empty).Replace("]", String.Empty)
```

Comment:
Yes, it's particularly strange since the format specifies that keywords should be escaped. I'll fix it.

---
In reply to: [74128521](https://github.com/dotnet/roslyn/pull/13037#discussion_r74128521) [](ancestors = 74128521)